services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/file/FileHandleManagerImplTest.java

```diff
@@ -807,11 +809,14 @@ public void testGetURLRequestWithAssociateAuthCheck() throws Exception {
 				.withAssociation(association.getAssociateObjectType(), association.getAssociateObjectId());
 		
 		String redirectURL = manager.getRedirectURLForFileHandle(request);
-		
 		verify(mockAuthorizationManager, times(1)).canDownLoadFile(mockUser, associations);
 		// Verifies that download stats are sent
-		verify(mockStatisticsCollector, times(1)).collectEvent(any());
-		
+		verify(messenger, times(1)).publishMessageAfterCommit(fileEventCaptor.capture());
+		FileEvent fileEvent = fileEventCaptor.getValue();
```

Comment:
why not use assertEquals with an expected file event? We can verify and sync the timestamp after the capture